core/src/main/java/org/apache/iceberg/BaseMetastoreTables.java

```diff
@@ -102,12 +100,12 @@ public Transaction beginReplace(Schema schema, PartitionSpec spec, Map<String, S
       return BaseTransaction.replaceTableTransaction(ops, metadata);
     } else {
       String location = defaultWarehouseLocation(conf, database, table);
-      metadata = newTableMetadata(ops, schema, spec, location, properties);
+      metadata = TableMetadata.newTableMetadata(ops, schema, spec, location, properties);
       return BaseTransaction.createTableTransaction(ops, metadata);
     }
   }
 
-  protected String defaultWarehouseLocation(Configuration conf,
+  protected static String defaultWarehouseLocation(Configuration conf,
```

Comment:
This is purposely not static so that implementations can override it. Our metastore tracks base locations for databases and our implementation uses those base locations to create new table paths.